### PR TITLE
[wasm] Unblock WBT that are not hanging anymore

### LINKED ISSUE #97054: [wasm] WBT tests using `BrowserRunner` seem to hang on a subsequent `dotnet new`
WBT tests failing on https://github.com/dotnet/runtime/pull/95980 . The pattern seems to be a test that uses `BrowserRunner`, then a subsequent test that uses `ToolCommand` (usually for `dotnet new`) gets stuck on waiting for the process to exit.

Notes from some investigation:
- `BrowserRunner` uses `ToolCommand` to start a process for `dotnet run`, but never explicitly tries to close that.
- this might be due to async completions not being able to post due to threadpools blocked?
- There are a couple of `Wait` and `task.Result` calls which also block
- Making the whole chain more async friendly, and killing the process in `Dispose` seems to help, but there is something still left

Note that these are being hit only on https://github.com/dotnet/runtime/pull/95980, and not on main.

### PR DESCRIPTION
**Pull request by ilonatommy**
Fixes https://github.com/dotnet/runtime/issues/97054. Unblocks tests that are passing when tested locally.

After temporarily reverting the changes in .yml files, to restore parallel runs, the CI still does not fail.

### CODE REVIEW

File: src/mono/wasm/Wasm.Build.Tests/Common/ToolCommand.cs [hunk 1]
```diff
@@ -133,7 +133,7 @@ private async Task<CommandResult> ExecuteAsyncInternal(string executable, string
             CurrentProcess.BeginOutputReadLine();
             CurrentProcess.BeginErrorReadLine();
             await completionTask;
-
+            CurrentProcess.Kill();
```
Comment (pavelsavara) [inline src/mono/wasm/Wasm.Build.Tests/Common/ToolCommand.cs:136]: I imagine that some child processes would be able to shutdown gracefully. 
Could we `await` graceful shutdown with timeout, say `1000`ms and only kill if that didn't resolve in time ?

Comment (lewing) [reply src/mono/wasm/Wasm.Build.Tests/Common/ToolCommand.cs:136]: Yeah, I would like to understand both why the process isn't exiting and why that is causing the next run to hang.  Is there a problem in the msbuild or compiler services?

Comment (radical) [reply src/mono/wasm/Wasm.Build.Tests/Common/ToolCommand.cs:136]: `completionTask` should really be completing only when the `Exited` event has been raised. You could add a `await CurrentProcess.WaitForExitAsync(cancellationToken)` after that to be sure we complete that side.

Also, I would suggesting dumping the process status and debugging this a bit. When I was looking at it, there was some odd behavior.

Comment (ilonatommy) [reply src/mono/wasm/Wasm.Build.Tests/Common/ToolCommand.cs]: CI does not fail anymore, even with parallel tests runs. We should unblock the tests.

**Review by pavelsavara — APPROVED** [review]

**Pull request merged**